test/bondingEvent.js

```diff
@@ -332,90 +339,129 @@ describe('BondingEvent', async () => {
         const expectedLiquidity = initialLiquidityTotal.mul(2);
         expect(position.liquidity).to.equal(expectedLiquidity);
       });
+
+      it('creates a second position if the price moves after initialisation', async () => {
+        // will create the first position
+        let amountSEuro = etherBalances.TWO_MILLION;
+        let { amountOther } = await BondingEvent.getOtherAmount(amountSEuro);
+        await SEuro.connect(customer).approve(BondingEvent.address, amountSEuro);
+        await USDT.connect(customer).approve(BondingEvent.address, amountOther);
+        await BondingEvent.connect(owner).bond(
+          customer.address, amountSEuro, durations.ONE_YR_WEEKS, rates.TEN_PC,
+        );
+
+        let positions = await BondingEvent.getPositions();
+        expect(positions).to.be.length(1);
+        let position = await BondingEvent.getPositionData(positions[0]);
+        expect(position.lowerTick).to.equal(pricing.lowerTick);
+        expect(position.upperTick).to.equal(pricing.upperTick);
+        expect(position.liquidity).to.be.gt(0);
+
+        // move the price to the edge of default tick range
+        // moves current price tick to 909
+        await swapSEuro(etherBalances['100K'].mul(5));
+
+        // bonding again will create a new position, as first one is not viable since price change
+        amountSEuro = etherBalances.TWO_MILLION;
+        ({ amountOther } = await BondingEvent.getOtherAmount(amountSEuro));
+        await SEuro.connect(customer).approve(BondingEvent.address, amountSEuro);
+        await USDT.connect(customer).approve(BondingEvent.address, amountOther);
+        await BondingEvent.connect(owner).bond(
+          customer.address, amountSEuro, durations.ONE_YR_WEEKS, rates.TEN_PC,
+        );
+
+        positions = await BondingEvent.getPositions();
+        expect(positions).to.be.length(2);
+        let secondPosition = await BondingEvent.getPositionData(positions[1]);
+        // since swap, new tick is at 909 - shifting lower tick to -700 and upper to 3300 puts 909 at in middle 20%
```

Comment:
that's just from experimenting with swapping an amount, and reading the ticks. again it's so dependent on the price changing every time it shifts to a new tick, that it would be complicated to have a calculation in the test.

i did some trial and error swapping a certain amount, until it was enough to test the behaviour i wanted. then i added comments to try to make the process a bit clearer